remixer_core/src/main/java/com/google/android/libraries/remixer/RemixerItem.java

```diff
@@ -39,20 +39,14 @@
    */
   private final int layoutId;
   /**
-   * A weak reference to the object that created this RemixerItem.
+   * A weak reference to this RemixerItem's context. The RemixerItem's lifecycle is tied to its
+   * contexts'.
    *
    * <p>It should be a reference to an activity, but it isn't since remixer_core cannot depend on
    * Android classes. It is a weak reference in order not to leak the activity accidentally.
    */
   @SuppressWarnings("unchecked")
-  private final WeakReference parentObject;
-  /**
-   * A copy of the parent object's class object. This will be necessary to know whether an object
-   * is of the same class as the parent object, even after the parent object has been reclaimed by
-   * the Garbage Collector.
-   */
-  @SuppressWarnings("unchecked")
-  private final Class parentObjectClass;
+  private final WeakReference context;
```

Comment:
i think for clarity you should use `WeakReference<Object>`. you may be able to remove the @SuppressWarnings here and below